### DISCUSSION TITLE
v0.6.0

### DISCUSSION THREAD

**Discussion by jmoralez** [Nixtla/nixtla#465] [Announcements]
# Breaking changes

* Removed `TimeGPT` class in favor of `NixtlaClient`.
* Removed `NixtlaClient.validate_token` method in favor of `NixtlaClient.validate_api_key`.
* Removed `short-horizon` and `long-horizon` models in favor of `timegpt-1` and `timegpt-1-long-horizon` respectively.
* Removed `fewshot_steps` and `fewshot_loss` in favor of `finetune_steps` and `finetune_loss` respectively.
* Removed `TIMEGPT_TOKEN` environment variable in favor of `NIXTLA_API_KEY`.
* Timestamps in the `time_col` preserve their type in the outputs (timestamp), previously they were cast to string.
* Gaps in series are not filled anymore, in line with [our documentation](https://nixtlaverse.nixtla.io/nixtla/docs/getting-started/data_requirements.html#important-considerations). Please use the [fill_gaps function](https://nixtlaverse.nixtla.io/utilsforecast/preprocessing.html#fill-gaps) if you require to do so.
* `NixtlaClient.weights_x` is now a list of lists if `num_partitions != None`, where each element corresponds to the weights for a specific partition.

# Features
* feat: call v2 endpoints by @jmoralez in https://github.com/Nixtla/nixtla/pull/429
* feat: orjson serialization by @jmoralez in https://github.com/Nixtla/nixtla/pull/452
* [FEAT] Add historical exogenous by @elephaint in https://github.com/Nixtla/nixtla/pull/453
* FEAT: Add feature contributions as attribute and tutorial on how to use by @marcopeix in https://github.com/Nixtla/nixtla/pull/448
* [FEAT] - Raise error when payload is too large and suggest number of partitions by @marcopeix in https://github.com/Nixtla/nixtla/pull/456

# Bug fixes
* fix: removed data input restriction during cross validation finetune by @Yibei990826 in https://github.com/Nixtla/nixtla/pull/426

# Enhancements
* feat: use TypeVar for dataframes by @jmoralez in https://github.com/Nixtla/nixtla/pull/443
* feat: set supported_models using base_url by @jmoralez in https://github.com/Nixtla/nixtla/pull/450
* feat: suggest earlier version for azure endpoints by @jmoralez in https://github.com/Nixtla/nixtla/pull/457

# Documentation
* updating excel docs with some more information by @tracykteal in https://github.com/Nixtla/nixtla/pull/414
* updating FAQ for fee information and Azure being available by @tracykteal in https://github.com/Nixtla/nixtla/pull/412
* Update 03_excel_addin.ipynb by @tracykteal in https://github.com/Nixtla/nixtla/pull/416
* [DOC] - Rephrase the sentence for specific loss function by @marcopeix in https://github.com/Nixtla/nixtla/pull/436
* docs(readme): getting started by @mergenthaler in https://github.com/Nixtla/nixtla/pull/438
* Adding activate trial link in intro by @tracykteal in https://github.com/Nixtla/nixtla/pull/417
* [DOC] - Add instructions on saving figures when not working in notebooks by @marcopeix in https://github.com/Nixtla/nixtla/pull/454
* [BUG]: close img tag in introduction notebook by @marcopeix in https://github.com/Nixtla/nixtla/pull/441
* docs: add polars quickstart by @jmoralez in https://github.com/Nixtla/nixtla/pull/447

**Full Changelog**: https://github.com/Nixtla/nixtla/compare/v0.5.2...v0.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Nixtla/nixtla/releases/tag/v0.6.0'>v0.6.0</a>.</em>
